hashprobe/aeshash/aeshash_test.go

```diff
@@ -4,43 +4,19 @@ import (
 	"encoding/binary"
 	"testing"
 	"time"
-	"unsafe"
 )
 
-//go:noescape
-//go:linkname runtime_memhash32 runtime.memhash32
-func runtime_memhash32(data unsafe.Pointer, seed uintptr) uintptr
-
-//go:noescape
-//go:linkname runtime_memhash64 runtime.memhash64
-func runtime_memhash64(data unsafe.Pointer, seed uintptr) uintptr
-
-//go:noescape
-//go:linkname runtime_memhash runtime.memhash
-func runtime_memhash(data unsafe.Pointer, seed, size uintptr) uintptr
-
-func memhash32(data uint32, seed uintptr) uintptr {
-	return runtime_memhash32(unsafe.Pointer(&data), seed)
-}
-
-func memhash64(data uint64, seed uintptr) uintptr {
-	return runtime_memhash64(unsafe.Pointer(&data), seed)
-}
-
-func memhash128(data [16]byte, seed uintptr) uintptr {
-	return runtime_memhash(unsafe.Pointer(&data), seed, 16)
-}
-
 func TestHash32(t *testing.T) {
 	if !Enabled() {
 		t.Skip("AES hash not supported on this platform")
 	}
 
-	h0 := memhash32(42, 1)
-	h1 := Hash32(42, 1)
+	testingInitAesKeySched()
```

Comment:
Could we call this from an `init` function in the test file so we don't have to repeat it in each test?

Reply:
I don't think we have any guarantees that the testing `init` function will be called before the other `init` function which sets `aeskeysched` too which is why I did it this way even though I don't like it either. Will think if there's another way to do it.

Reply:
Ah! Because the tests are in the same package. 

We could enforce it by moving tests to a aeshash_test package but that seems a bit overkill, the current approach is good enough 👍

Reply:
I just modified it to check `aeskeysched` before initialization in the `amd64.go` file which I think should work regardless of `init` order.